src/psyclone/psyir/nodes/node.py

```diff
@@ -891,6 +891,22 @@ def _insert_schedule(self, children=None, ast=None):
             sched.ast = ast
         return sched
 
+    def find_symbol_table(self):
+        '''
+        :returns: the symbol table attached to the nearest ancestor \
+            node (including self).
+        :rtype: :py:class:`psyclone.psyir.symbols.SymbolTable`
+
+        :raises InternalError: if no symbol table is found.
+
+        '''
+        current = self
+        while current and not hasattr(current, "symbol_table"):
+            current=current.parent
+        if current:
+            return current.symbol_table
+        raise InternalError("Symbol table not found in any ancestor nodes.")
+        
```

Comment:
pylint: trailing whitespace.

Reply:
Oops. I thought I'd tested all of these (sorry).